src/ui/components/MapBox/MapBox.js

```diff
@@ -0,0 +1,222 @@
+ui.class.Mapbox = class Mapbox extends HTMLElement {
+	constructor(props) {
+		super();
+		const self = this;
+
+		let m_parentTag;
+		let m_containerTag;
+
+		let m_props;
+		let m_map;
+
+		this.addPoints = addPoints;
+		this.setup = setup;
+
+		if (props) {
+			setup(props);
+		}
+
+		async function setup(props) {
+			await configure(props);
+			await addApiToDocument();
+			await ui.utils.installUIComponent({ self, m_parentTag, m_props });
+			await setupDOM();
+			await setMapBoxAccessToken();
+			await renderMap();
+			await getUserLocation();
+			setupComplete();
+		}
+
+		function addApiToDocument() {
+			if (!m_props.apis || !m_props.apis.length) {
+				throw "Please, add apis prop to run MapBox";
+			}
+
+			return new Promise((resolve) => {
+				let apisLoaded = 0;
+				const apisTotal = m_props.apis.length * 2;
+
+				ui.d.append("head", "<!-- Load Mapbox Libraries -->");
+
+				m_props.apis.forEach((api, index) => {
+					let script;
+					let link;
+					let arr = [];
+
+					if (api.js) {
+						script = ui.d.createTag("script", {
+							prop: { src: api.js },
+							attr: { "data-mapboxscripts": `js-${index}` }
+						});
+						ui.d.append("head", script);
+						arr.push(script);
+					}
+					if (api.css) {
+						link = ui.d.createTag("link", {
+							prop: {
+								href: api.css,
+								rel: "stylesheet"
+							},
+							attr: { "data-mapboxscripts": `css-${index}` }
+						});
+						ui.d.append("head", link);
+						arr.push(link);
+					}
+
+					arr.forEach(tag => {
+						tag.addEventListener("load", () => {
+							apisLoaded++;
+							if (apisLoaded === apisTotal) {
+								resolve();
+							}
+						});
+					});
+				});
+
+				ui.d.append("head", "<!-- End Mapbox Libraries -->");
+
+			});
+		}
+
+		function setMapBoxAccessToken() {
+			return new Promise((resolve) => {
+				mapboxgl.accessToken = m_props.access_token;
```

Comment:
I missed this one before, it should be moved into configure{} as well and delete the function altogether